### PR TITLE
Move to elasticsearch-java 8

### LINKED ISSUE #305: Add elasticsearch 8.x support
### Feature description

In the new version, the client for requests to elastic has slightly changed. Documentation can be found here:
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/migrate-hlrc.html

### PR DESCRIPTION
**Pull request by altro3**
Removed the use of RestHighLevelClient and xcontent

Fixes #305

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,12 +1,12 @@
 projectVersion=4.2.1-SNAPSHOT
```
Comment (graemerocher) [inline gradle.properties:1]: Can you bump this to a new major version

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 1 above)
Comment (altro3) [reply gradle.properties]: @graemerocher yes, sure. Already done

**Comment by altro3** [comment]
@graemerocher sorry, I don't understand what does it mean:
```
Execution failed for task ':elasticsearch:japiCmp'.
> A failure occurred while executing me.champeau.gradle.japicmp.JApiCmpWorkAction
   > Detected binary changes.
         - current: elasticsearch-5.0.0-SNAPSHOT.jar
         - baseline: micronaut-elasticsearch-4.2.0.jar.
```
How to fix it?

**Comment by graemerocher** [comment]
This is because the changes are binary incompatible with the previous release. To resolve this the changes have to accepted with a file like:

https://github.com/micronaut-projects/micronaut-core/blob/3.4.x/config/accepted-api-changes.json

You can run `./gradlew elasticsearch:japiCmp` and open the report to see example JSON snippets that need to be included in this PR to accept the changes.

We have this in place to prevent breaking binary incompatibility in projects

**Comment by altro3** [comment]
@altro3 thanks! Now all fixed

**Review by graemerocher — APPROVED** [review]

**Comment by graemerocher** [comment]
@timyates This module upgrade features a breaking major version upgrade so we would only be able to include in the BOM from Micronaut 4.0 onwards

**Comment by timyates** [comment]
@graemerocher should this be 4.0.x?

https://github.com/micronaut-projects/micronaut-elasticsearch/pull/320/files#diff-3d103fc7c312a3e136f88e81cef592424b8af2464c468116545c4d22d6edcf19R17

**Comment by graemerocher** [comment]
it should indeed

**Review by timyates — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+elasticsearchVersion=8.2.0
 
 title=Micronaut Elasticsearch
 projectDesc=Integration between Micronaut and Elasticsearch
 projectUrl=https://micronaut.io
 githubSlug=micronaut-projects/micronaut-elasticsearch
 developers=Puneet Behl
 
-githubCoreBranch=3.3.x
+githubCoreBranch=3.5.x
```
[the remark this answers is not in the document]
Comment (altro3) [reply gradle.properties]: done

**Review by timyates — APPROVED** [review]

**Pull request merged**